src/movies.js

```diff
@@ -1,16 +1,170 @@
+        let roundedAverage = (averageRates.toFixed(2));
+        roundedAverage = Number(roundedAverage); 
+        return roundedAverage
+
+    } else {
+        return 0; 
+    }
+    }
+
+
+
+
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+const dramaMoviesRate = (array) => {
+    if (array.genre !== "Drama") {
+        return 0;
+    }; 
+
+    let dramaMovies = array.filter((movie) => {
+        return movie.genre.includes("Drama");
+     });
+
+    let dramaAverageRates = array.reduce((acc, i) => {
+            return acc + i.rate / dramaMovies.length
+        }, 0)
+
+    return dramaAverageRates
+}
+
+    //
+
+//     let dramaMovies = array.filter((movie) => {
+//         return movie.genre.includes("Drama");
+//     });
+    
+//     dramaAverageRates.reduce((acc, i) => {
+//         if (i.rate != '' && i.rate != undefined){
+//             return acc + i.rate
+//         } else {
+//             return acc + 0
+//         }
+//     }, 0);
+
+//     let averageDramaRates = (dramaMovies / array.length).toFixed(2);
+//     averageDramaRates = Number(averageDramaRates); 
+//     return averageDramaRates;
+// }
+
+
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = (array) => {
+
+    // let a = JSON.stringify(array)
+    // let b = JSON.parse(a)
+
+    let movieByYear = JSON.parse(JSON.stringify(array))
+    //let movieByYear2 = array
+
+
+    movieByYear.sort((a, b) => {
+        if (a.year > b.year) {
+                    return 1;
+                }
+                else if (a.year < b.year) {
+                  return -1;
+                }
+                else {
+                    if (a.title > b.title) {
+                        return 1;
+                    }
+                    else if (a.title < b.title) {
+                      return -1;
+                    }
+                    else {
+                      return 0
+                    }
+                }
+    });
+    return movieByYear;
+};
+
+
+
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
 
+const orderAlphabetically = (array) => {
+
+    let orderedMovies = JSON.parse(JSON.stringify(array))
+    let orderedTitles = orderedMovies
+
+   let moviesSort = orderedTitles.sort((a, b) => {
+        if (a.title > b.title) {
+            return 1;
+        }
+        else if (a.title < b.title) {
+          return -1;
+        }
+        else {
+            if (a.title > b.title) {
+                return 1;
+            }
+            else if (a.title < b.title) {
+              return -1;
+            }
+            else {
+              return 0
+            }
+        }
+
+        // if (orderedTitles.length > 21) {
+        //     return orderedTitles.splice(0, 21)
+        // } else {
+        //     return orderedTitles;
+        // }
+
+    });
+
+    let moviesTitle = orderedTitles.map((elem) => {
+        return elem.title
+   }) 
```

Comment:
Good approach this far! the only thing you were missing was getting the first 20 elements, you can do this with the .slice method for example. So in the end of the function, just returning moviesTitle.slice(0, 20)